Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixMechanical.h

```diff
@@ -0,0 +1,1514 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+/******************************************************************************
+* Contributors:
+*   - InSimo
+*******************************************************************************/
+#pragma once
+
+#include <sofa/linearalgebra/config.h>
+#include <sofa/linearalgebra/CompressedRowSparseMatrixGeneric.h>
+
+#include <sofa/type/trait/is_vector.h>
+
+namespace sofa::linearalgebra
+{
+
+/// Mechanical policy type, showing the types and flags to give to CompressedRowSparseMatrixMechanical
+/// for its second template type.
+class CRSMechanicalPolicy : public CRSDefaultPolicy
+{
+public:
+    static constexpr bool CompressZeros = true;
+    static constexpr bool IsAlwaysSquare = true;
+    static constexpr bool IsAlwaysSymmetric = true;
+    static constexpr bool OrderedInsertion = true;
+    static constexpr bool StoreLowerTriangularBlock = true;
+
+    static constexpr int matrixType = 1;
+};
+
+class CRSMechanicalStoreTouchedPolicy : public sofa::linearalgebra::CRSDefaultPolicy
```

Comment:
Is it used?

Reply:
ISSofa uses it for the class sofa::core::BlocMatrixWriter.h, which is not really used in our code base but resembles what you have done here and there to optimize block writing (as far as I could see, I could be wrong).
Anyway we can still remove it for now, and add it later if necessary if you want